### PR TITLE
CLOUD-360 One shot deployment to OpenShift

### PR DESCRIPTION
**Pull request by cap1984**
Some short info regarding the image build.
The image has to be assembled via the ./e2e-tests/build script
$ IMAGE=<image-name> ./e2e-tests/build 
$ docker run -v /tmp/fake_home:/tmp/dbaas -it DBAAS-cli percona-dbaas gcloud gcloud setup <env> --gcloud-cluster <cluster> --gcloud-zone <zone> --gcloud-project <project> --kube-namespace <namespace> --gcloud-key-file <filepath>

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -17,5 +17,10 @@ RUN curl -o /licenses/LICENSE.kubectl https://raw.githubusercontent.com/kubernet
 RUN cp /usr/share/google-cloud-sdk/LICENSE /licenses/LICENSE.gcloud
 COPY build/_output/bin/percona-dbaas /usr/local/bin/percona-dbaas
 
+RUN mkdir /tmp/dbaas
```
Comment (delgod) [inline build/Dockerfile]:
```suggestion
RUN install -d -o 99 -g 0 -m 0775 /tmp/dbaas
```

**Review by delgod — CHANGES_REQUESTED** [review]

File: build/Dockerfile (hunk 1 above)
Comment (cap1984) [reply build/Dockerfile]: done

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:master-broker-PR-44-604655e

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:master-broker-PR-44-61f16cd

**Review by nonemax — APPROVED** [review]

**Review by dAdAbird — APPROVED** [review]

File: deploy/percona-broker.yaml [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+   port: 8081
+   targetPort: 8081
+---
+apiVersion: servicecatalog.k8s.io/v1beta1
+kind: ClusterServiceBroker
+metadata:
+  name: percona-service-broker
+  namespace: percona-service-broker
+spec:
+  url: 'http://percona-service-broker.percona-service-broker.svc:8081'
```
Comment (delgod) [inline deploy/percona-broker.yaml]:
```suggestion
  url: 'http://percona-service-broker.percona-service-broker.svc.cluster.local:8081'
```

File: deploy/percona-broker.yaml [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+  name: percona
+subjects:
+  - kind: ServiceAccount
+    name: percona-service-broker
+    namespace: percona-service-broker
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: percona
```
Comment (delgod) [inline deploy/percona-broker.yaml]:
```suggestion
  name: percona-service-broker
```

File: deploy/percona-broker.yaml [hunk 4]
```diff
@@ -0,0 +1,557 @@
...
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: percona-service-broker
+  namespace: percona-service-broker
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: percona
```
Comment (delgod) [inline deploy/percona-broker.yaml]:
```suggestion
  name: percona-service-broker
```

File: deploy/percona-broker.yaml [hunk 5]
```diff
@@ -0,0 +1,557 @@
...
+subjects:
+  - kind: ServiceAccount
+    name: percona-xtradb-cluster-operator
+    namespace: percona-service-broker
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: percona-xtradb-cluster-operator
+  namespace: percona-service-broker
```
Comment (delgod) [inline deploy/percona-broker.yaml]: Roles should be created from CLI tool

File: deploy/percona-broker.yaml [hunk 6]
```diff
@@ -0,0 +1,557 @@
...
+  - update
+  - patch
+  - delete
+  - deletecollection
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: percona-server-mongodb-operator
+  namespace: percona-service-broker
```
Comment (delgod) [inline deploy/percona-broker.yaml]: is the broker can run operator in different namespaces?

File: deploy/percona-broker.yaml [hunk 7]
```diff
@@ -0,0 +1,557 @@
...
+      app: percona-service-broker
+  template:
+    metadata:
+      labels:
+        app: percona-service-broker
+    spec:
+      serviceAccountName: percona-service-broker
+      containers:
+        - name: percona-service-broker
+          image: perconalab/percona-xtradb-cluster-operator:pxc-service-broker
```
Comment (delgod) [inline deploy/percona-broker.yaml]:
```suggestion
          image: percona/percona-xtradb-cluster-operator:1.1.0-broker
```

File: deploy/percona-broker.yaml [hunk 8]
```diff
@@ -0,0 +1,557 @@
...
+  - get
+  - list
+  - update
+  - watch
+  - create
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  - replicasets
```
Comment (delgod) [inline deploy/percona-broker.yaml:87]: is broker creates 'replicasets' or 'statefulsets' or 'jobs' or 'cronjobs' or 'certmanager' ?

File: deploy/percona-broker.yaml [hunk 9]
```diff
@@ -0,0 +1,557 @@
...
+      type: string
+      description: Job status
+      JSONPath: .status.state
+    - name: Completed
+      description: Completed time
+      type: date
+      JSONPath: .status.completed
+    - name: Age
+      type: date
+      JSONPath: .metadata.creationTimestamp
```
Comment (delgod) [inline deploy/percona-broker.yaml]: what about PSMDB CRDs?

**Review by delgod — CHANGES_REQUESTED** [review]

File: deploy/percona-broker.yaml (hunk 2 above)
Comment (cap1984) [reply deploy/percona-broker.yaml]: done

File: deploy/percona-broker.yaml (hunk 3 above)
Comment (cap1984) [reply deploy/percona-broker.yaml]: done

File: deploy/percona-broker.yaml (hunk 9 above)
Comment (cap1984) [reply deploy/percona-broker.yaml]: PXC crd has been removed

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:master-broker-PR-44-2169947

File: deploy/percona-broker.yaml (hunk 5 above)
Comment (cap1984) [reply deploy/percona-broker.yaml]: done

File: deploy/percona-broker.yaml (hunk 6 above)
Comment (cap1984) [reply deploy/percona-broker.yaml]: nope, the ticket has been created

File: deploy/percona-broker.yaml (hunk 8 above)
Comment (cap1984) [reply deploy/percona-broker.yaml:87]: here is what we get from Kubernetes when psmdb or pxc roles has been applied


```
Error from server (Forbidden): error when creating "psmdbrole.yaml": roles.rbac.authorization.k8s.io "percona-server-mongodb-operator" is forbidden: user "system:serviceaccount:percona-service-broker-3:percona-service-broker" (groups=["system:serviceaccounts" "system:serviceaccounts:percona-service-broker-3" "system:authenticated"]) is attempting to grant RBAC permissions not currently held:
{APIGroups:[""], Resources:["persistentvolumeclaims"], Verbs:["get" "list" "watch" "create" "update" "patch" "delete"]}
{APIGroups:[""], Resources:["pods"], Verbs:["get" "list" "watch" "create" "update" "patch" "delete"]}
{APIGroups:[""], Resources:["pods/exec"], Verbs:["get" "list" "watch" "create" "update" "patch" "delete"]}
{APIGroups:[""], Resources:["secrets"], Verbs:["get" "list" "watch" "create" "update" "patch" "delete"]}
{APIGroups:[""], Resources:["services"], Verbs:["get" "list" "watch" "create" "update" "patch" "delete"]}
{APIGroups:["batch"], Resources:["cronjobs"], Verbs:["get" "list" "watch" "create" "update" "patch" "delete"]}
{APIGroups:["certmanager.k8s.io"], Resources:["certificates"], Verbs:["get" "list" "watch" "create" "update" "patch" "delete" "deletecollection"]}
{APIGroups:["certmanager.k8s.io"], Resources:["issuers"], Verbs:["get" "list" "watch" "create" "update" "patch" "delete" "deletecollection"]}
{APIGroups:["policy"], Resources:["poddisruptionbudgets"], Verbs:["get" "list" "watch" "create" "update" "patch" "delete"]}
```

ClusterRole shall contain all those permissions for inheritance

File: deploy/percona-broker.yaml (hunk 7 above)
Comment (cap1984) [reply deploy/percona-broker.yaml]: 1.2.0

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:master-broker-PR-44-8660870

**Review by delgod — APPROVED** [review]

**Pull request merged**